microsoft_teams/icon_microsoft_teams/triggers/new_message_received/trigger.py

```diff
@@ -105,10 +105,22 @@ def get_sorted_messages(self, messages_endpoint: str) -> list:
         messages_result = requests.get(messages_endpoint, headers=headers)
         try:
             messages_result.raise_for_status()
+
+
+        # The beta API bombs out every once in a while with Auth denied. Try forcing a refersh of our auth token and
+        # retry getting messages before raising an exception.
         except Exception as e:
```

Comment:
Is there a specific exception that is returned when the token expires that could be caught explicitly? Then the refresh and retry would only be run in situations where it is necessary.

Reply:
No. It's a 400 unauthorized. The beta isn't fleshed out and there are a few things that can cause that. The other issue, if I want to pull the error message, I'll have to have exception handling in my exception handling, and another exception wrapping that. I really don't want to go down the exceptionception path.

Reply:
I feel that adding more/better exception handling is preferable to sending a second request unnecessarily.

Reply:
To Tyler's point, I think it's better to ask what is the best, not most convenient, way to solve the problem for the customer. If we can provide better messaging/logging by getting to the core error message then that seems like it's the answer.

Reply:
I would put better messaging around this if I could. But right now the errors popping out here aren't well defined. Also, we have no clue why this call fails, my bet right now is the Beta is being a beta and not fully fleshed out. 

In the face of ill-defined errors, and unknown causes. I think this the best I can do around this error handling.